### DISCUSSION TITLE
Set up external service profile for ingress traffics

### DISCUSSION THREAD

**Discussion by taind772** [linkerd/linkerd2#9498] [Help]
Due to some reasons, I couldn't use the Linkerd proxy in Ingress mode for Ingress controllers. I tried to set up an external service profile for a service using the host header (domain name in case of ingress controller), but the config didn't seem to work well.

Currently, I have:
- Kong Ingress controller with Linkerd proxy enabled (default proxy, not in the Ingress mode)
- A service (ex: `echo-kong-server.lab`) which have Linkerd proxy meshed too
- Two service profiles, all in lab namespace:
     - echo-kong-server.lab.svc.cluster.local
     - echo-kong.xxx.net (the public domain)

I tried to enable external-service-profiles on both the ingress controller & the pod, but still, there weren't any `route_.*` metrics.

The response from the echo server:
```
Request served by echo-kong-server-6c85f87c4-6h4h7

HTTP/1.1 GET /

Host: echo-kong.xxx.net
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.5
Grpc-Metadata-Request-Id: 425970ef-97dd-4866-ab20-81a49a6e47de#13
L5d-Client-Id: ingress-kong.ingress-kong.serviceaccount.identity.linkerd.xxx.cluster.local
Sec-Fetch-Dest: document
Sec-Fetch-Mode: navigate
Sec-Fetch-Site: none
Sec-Fetch-User: ?1
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0
X-Forwarded-For: {{myIp}}
X-Forwarded-Host: echo-kong.xxx.net
X-Forwarded-Path: /
X-Forwarded-Port: 443
X-Forwarded-Proto: https
X-Real-Ip: {{myIp}}
```

Is it possible to get Per-route metrics from ingress traffic without using the Linkerd proxy in Ingress mode?

**Comment by adleong** [comment]
Hi @ndtai772!  Unfortunately, inbound traffic into the kong ingress controller via the public domain is HTTPS which means that the Linkerd proxy can't decrypt it and apply service profiles.

**Reply by taind772** [reply]
Hi @adleong , thanks for your reply.
Actually, I mean: can I export per-route metrics (by service profile) of the traffic from the Ingress controller to Kubernetes services (which was HTTP) without using  Linkerd in Ingress mode?

The request
![image](https://user-images.githubusercontent.com/49815011/193132588-0e15e3ec-e321-496d-9616-165227382160.png)


I have tried setup service profiles:
![image](https://user-images.githubusercontent.com/49815011/193133037-9ca094e4-6ef2-409f-a64a-d34c47d829ee.png)

![image](https://user-images.githubusercontent.com/49815011/193133372-5f2b6d2b-baca-445d-aa35-0c4e72012cdc.png)

**Reply by adleong** [reply]
Where are you looking for the `route_.*` metrics?  Keep in mind that outbound metrics are separate from inbound metrics.
